Dockerfile.template

```diff
@@ -1,23 +1,59 @@
-FROM balenalib/%%BALENA_MACHINE_NAME%%-ubuntu-python:3.8.13-impish
+FROM ubuntu:jammy-20220531 as kernel-build
 
-RUN install_packages \
-    curl \
-    wget \
-    build-essential \
-    libelf-dev \
+ARG DEBIAN_FRONTEND=noninteractive
+ENV TZ=Etc/UTC
+
+RUN apt-get update && \
+    apt-get install -y \
     awscli \
     bc \
+    bison \
+    build-essential \
+    curl \
     flex \
+    git \
+    libelf-dev \
     libssl-dev \
-    bison
+    wget
 
+WORKDIR /usr/src/app
 
-COPY . /usr/src/app
+# Clones the kernel builder repository
+RUN git clone https://github.com/balena-os/kernel-module-build.git .
 
-WORKDIR /usr/src/app
+# You can switch to a specific commit of the kernel builder if you want to ensure consistent builds
+# git reset --hard f889851c5d00890b51b27c63eaa59a976063265a
+
+# Copy your kernel source from your local build context to the build directory
+COPY example_module .
+
+# Set the name of the directory where you have copied your kernel source in the step above
+ENV MOD_DIR "example_module"
+
+# Set the balena OS version you intend to use
+ENV OS_VERSION '2.99.27'
```

Comment:
Could this be a cloud configuration variable?

Reply:
Not sure I follow. Once it is deployed to a fleet, don't people then get to choose which OS they run it on? So if I join the fleet today I might get 2.99.27 and in a month something else by default? 

I thought the versioning of the OS isn't handled by the cloud and opened https://jel.ly.fish/search/pattern-allow-specifying-balena-os-version-used-when-downloading-joining-via-balena-hub-1d0d210@1.0.0

I think it would be good to have a way to pin a Cloud deploy to a specific OS version but wasn't sure if that functionality existed yet.

Reply:
hey @maggie0002 as I see it an application that builds an installs a kernel module is introducing a dependency on the hostOS so they cannot just update to whatever version. Ideally they would use a contract that specifies the dependency in a similar way to how the L4T dependencies are specified in the Jetson boards (see https://www.balena.io/docs/learn/develop/container-contracts/#board-support). And we could probably use that dependency as well to fill the OS version.

Reply:
It's just something that is not well-resolved at the moment either, just mentioning it so we can put some thought in how to make this dependency obvious

Reply:
Absolutely, the kernel modules get pinned to that OS yet we are not able to pin a cloud build to an OS release. By 'get to choose which OS they run it on' I meant in terms of this being a problem rather than a feature.

Going to ping @aethernet too who I think is working on the preloading which has implications (apologies if I pinged the wrong person).

Seems I do not have the permission to merge this one, but if you do then all LGTM.